packages/@react-spectrum/label/src/StaticField.tsx

```diff
@@ -0,0 +1,200 @@
+        </Label>}
+      <div
+        className={classNames(labelStyles, 'spectrum-Field-wrapper')}>
+        <div
+          className={classNames(labelStyles, 'spectrum-Field-field')}>
+          {Array.isArray(value) && 
+            // @ts-ignore
+            <FormattedStringList value={value} formatOptions={formatOptions as Intl.ListFormatOptions} />}
+
+          {typeof value === 'object' && 'start' in value && typeof value.start === 'number' && typeof value.end === 'number' && 
+            <FormattedNumber value={value as NumberValue} formatOptions={formatOptions as Intl.NumberFormatOptions}  />}
+
+          {typeof value === 'object' && 'start' in value && typeof value.start !== 'number' && typeof value.end !== 'number' && 
+            <FormattedDateRange value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'number' && 
+            <FormattedNumber value={value} formatOptions={formatOptions as Intl.NumberFormatOptions} />}
+
+          {(typeof value === 'object' && ('calendar' in value || 'hour' in value) || (value instanceof Date))  &&
+            <FormattedDate value={value} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'string' && 
+            props.value}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function FormattedStringList<T extends string[]>(props: StringListProps<T>) {
+  let stringFormatter = useListFormatter(props.formatOptions); 
+
+  return (
+    <>{stringFormatter.format(props.value)}</>
+  );
+}
+
+function FormattedNumber<T extends NumberValue>(props: NumberProps<T>) {
+  let numberFormatter = useNumberFormatter(props.formatOptions); 
+  let value = props.value;
+
+  if (typeof value === 'object' && typeof (value as RangeValue<NumberValue>).start === 'number' && typeof (value as RangeValue<NumberValue>).end === 'number') {
+    return <>{numberFormatter.formatRange(value.start, value.end)}</>;
+  }
+
+  return <>{numberFormatter.format(value as number)}</>;
+}
+
+function FormattedDateRange<T extends DateTimeValue>(props: DateProps<T>) {
+  let dateFormatter = useDateFormatter(props.formatOptions);
+  let value = props.value;
+  let timeZone = dateFormatter.resolvedOptions().timeZone || getLocalTimeZone();
+  let start;
+  let end;
+
+  if (typeof value === 'object' && 'start' in value) {
+    start = value.start;
+    end = value.end;
+
+    if (!(start instanceof Date) && !(end instanceof Date)) {
+      if (start instanceof Time && end instanceof Time) {
+        start = convertValue(start).toDate(timeZone);
+        end = convertValue(end).toDate(timeZone);
+      } else if (!(start instanceof Time) && !(end instanceof Time)) {
+        start = start.toDate(timeZone);
+        end = end.toDate(timeZone);
+      }
+    } 
+  }
+  
+  return <>{dateFormatter.formatRange(start, end)}</>;
+}
+
+function FormattedDate<T extends DateTimeValue>(props: DateProps<T>) {
```

Comment:
Is there a reason the Date and DateRange formatters are split into separate components? For numbers there is a single one that does both. Seems like there is a reasonable amount of the logic that could be shared. You could pull out some of the logic below into a reusable function that accepts a `DateTime` and returns a `Date`, and then reuse it for both the start and end of a range as well.